src/main.cpp

```diff
@@ -2,136 +2,114 @@
 #include <MFRC522.h>
 #include "SoftwareSerial.h"
 #include "DFRobotDFPlayerMini.h"
+#include <FastLED.h>
 
 // logic
 #define MY_LEVEL 1
 
 // RFID
 #define SS_PIN 10
 #define RST_PIN 9
-MFRC522 rfid(SS_PIN, RST_PIN); // Instance of the class
+MFRC522 mfrc522(SS_PIN, RST_PIN); // Instance of the class
+#include "MFRC522_func.h"
+
 MFRC522::MIFARE_Key key; 
 // Init array that will store new NUID 
 byte nuidPICC[4];
 byte blockAddr      = 4;
 byte buffer[18];
 byte size = sizeof(buffer);
 byte trailerBlock   = 7;
+bool read_success, write_success, auth_success;
```

Comment:
What is the motivation to make them global?
Can they be declared as function variables instead?

```cpp
bool auth_success = authenticate(trailerBlock, key);
if(!auth_success) {
  return;
}
```